client/my-sites/upgrades/checkout/thank-you.jsx

```diff
@@ -63,6 +66,14 @@ var CheckoutThankYou = React.createClass( {
 	},
 
 	componentDidMount: function() {
+		var selectedSite = this.props.lastTransaction.selectedSite;
+		this.props.refreshSitePlans( selectedSite.ID );
```

Comment:
You are right, we should refresh only when the user purchased a plan. I tested with themes though and it never reaches the thank you page.
I haven't tested domain or other upgrades so I guess it's cleaner to check for the type of transaction explicitly.